### PR TITLE
docs: update dynamic privileges proposal for SEM changes

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Problem Summary:

This updates the dynamic privileges proposal to incorporate some design changes / differences in the implementation to original proposal.

### What is changed and how it works?

Proposal update only.

### Related changes

- None

### Check List 

Tests 

- No code

Side effects

- None

### Release note 

- No release note

### CODE REVIEW

File: docs/design/2021-03-09-dynamic-privileges.md [hunk 1]
```diff
@@ -233,14 +233,16 @@ No change
 | `SYSTEM_VARIABLES_ADMIN` | Allows changing any GLOBAL system variable. | Currently this required `SUPER`. It will now require `SYSTEM_VARIABLES_ADMIN` or `SUPER`. |
 | `ROLE_ADMIN` | Allows granting and revoking roles. | Won’t allow revoking on restricted_users (see below). |
 | `CONNECTION_ADMIN` | Allows killing connections. | Like `PROCESS` static privilege, but slightly more restrictive (no show processlist). |
+| `SYSTEM_USER` | The user can't be altered or dropped by only the `CREATE USER` privilege. | Helps prevent security escalations. |
```
Comment (xhebox) [inline docs/design/2021-03-09-dynamic-privileges.md:236]: Any link to this change? I have no idea of that.

Comment (morgo) [reply docs/design/2021-03-09-dynamic-privileges.md:236]: That's right. It's not merged, but adding this one makes a lot of sense. See: https://mysqlserverteam.com/the-system_user-dynamic-privilege/

**Review by xhebox — APPROVED** [review]
LGTM

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**